pkg/cmd/repo/create/create.go

```diff
@@ -644,17 +675,7 @@ func localInit(gitClient *git.Client, remoteURL, path, checkoutBranch string) er
 		return err
 	}
 
-	if checkoutBranch == "" {
-		return nil
-	}
-
-	refspec := fmt.Sprintf("+refs/heads/%[1]s:refs/remotes/origin/%[1]s", checkoutBranch)
-	err = gc.Fetch(ctx, "origin", refspec)
```

Comment:
I sense a subtle change in functionality:

Previously using `--template` resulted in:
```
git init
git remote add origin <URL>
git fetch origin +refs/heads/MAIN:refs/remotes/origin/MAIN
```
Now it will be:
```
git clone <URL> --branch MAIN
```
The difference is that in the latter case, the `origin` remote will only be configured to fetch from a single branch, while the previous approach did a one-time fetch from a single fetch, but future fetches would be allowed to fetch all branches.

Reply:
Oh. I forgot that without `--single-branch`, `--branch` will not lock the remote configuration into fetching just a single branch. My bad, sorry!

Reply:
Minhanh@141

Reply:
> I sense a subtle change in functionality:
> 
> 
> 
> Previously using `--template` resulted in:
> 
> ```
> 
> git init
> 
> git remote add origin <URL>
> 
> git fetch origin +refs/heads/MAIN:refs/remotes/origin/MAIN
> 
> ```
> 
> Now it will be:
> 
> ```
> 
> git clone <URL> --branch MAIN
> 
> ```
> 
> The difference is that in the latter case, the `origin` remote will only be configured to fetch from a single branch, while the previous approach did a one-time fetch from a single fetch, but future fetches would be allowed to fetch all branches.
> 
>